### ISSUE TITLE
Mailing lists

### ISSUE THREAD

**Issue by r10s** [deltachat/deltachat-core#125]
After reworking the internal structure in PR #115 and #118, supporting, at least read-only, mailing lists should be much easier.

The current state is that messages belonging to mailing lists are detected and are explicitly hidden, see the comment ["Current state of mailing list handling" in the source](https://github.com/deltachat/deltachat-core/blob/master/src/mrmimeparser.c#L1723).

If we decide to support mailing lists - what is needed, what are the headers, what are the corner cases? How can they be shown and used in Delta Chat? [RFC 2919](https://tools.ietf.org/html/rfc2919) and [RFC 3834](https://tools.ietf.org/html/rfc3834) are very helpful, however, in practice things are often done differently and require the know-how from experienced people :)

Todo's that came into my mind for **reading mailing lists** - please comment below and add what i've forgotten:

- [ ] is the current detection of mailing lists sufficient?
- [ ] collect messages with the same `List-Id:` header in a separate chat
- [ ] check, if it is needed to normalize the `List-Id` (in `List-Id: foo <bla>` - which part in practice is the id? `foo <bla>` or `bla`?)
- [ ] check, what should be used as the name for the list - `foo` in the example above? or the `[prefix]` from the subject?
- [ ] do add senders/recipients of belonging messages as "known contacts"
- [ ] avoid replying to the mailing list

The messages will be shown in chronological order, this is not always optimal, however, I do not think that and extra view is worth the effort currently.

**Writing mailing lists** is more complicated:

- [ ] make sure replies go the `Reply-To:` header address (used eg. for the github issues) or to the `List-Post:` - is this correct? if both are unset - use `From:`?
- [ ] I think Delta Chat would need an explicit reply option so that threads are not messed up on the normal MUAs. The reply option should be invoked by a long press on the message to reply to. However, this feature is not only useful for mailing lists :)
- [ ] Check, how the subject should look like; I think the "Chat: bla bla bla" is not a perfect solutions for classical mailing lists, maybe we should go for a classical "Re: bla bla bla" (we can switch here as we want, the Chat:-prefix is not required anywhere for Delta Chat to work) Again, this feature is not only useful for mailing lists :)
- [ ] Set `In-Reply-To:` and `References:` carefully - how?
- [ ] In general, make sure, the Delta Chat replies are "okay" to the users using classical MUAs to read/write mailing lists
- [ ] Administration functions, Unsubscribe etc. seem to be out of scope for the first steps. OTOH, maybe a simple link to the correct page is sufficient.

Any help on these questions is very welcome. I know there are real mailing lists experts out there :)

**Comment by testbird** [comment]
`[prefix]` is probaly a usefull short name for the group, here too, and `foo` maybe usefull as description?

**Comment by testbird** [comment]
> Delta Chat would need an explicit reply option ...invoked by a long press on the message to reply to

In (managed) mailing list groups it could probably be assumed that replies default to go to the `reply-to`. (Whereas in peer2peer recipient list groups the reply defaults to go out to everybody.)

**Comment by r10s** [comment]
Yes, the relies would go to In-Reply-To, if available, but this has nothing to do with an reply-option which would set In-Reply-To (any maybe References)

**Comment by testbird** [comment]
You mean an option to reply to specific messages in the chat, instead of sending a new message?
I didn't get yet why that would require a long press.

**Comment by r10s** [comment]
If you want to reply to a specifiy message, you have to select it in a way. 
- The long clock currently shows _Forward_ and _Delete_ buttons currently; there could be an additional _Reply_ button.
- Tapping on the _Reply_ button would select the message for a reply, this can be shown as an hint somewhere
- Now you can compose the message as usual, when sending it, it will be send as a correct reply for normal MUAs (but maybe also Delta Chat could show this somewhere).

WhatsApp, Telegram and other offer a similar functionality; so this feature won't be new to users that have used another messengers before.

**Comment by testbird** [comment]
I see. What I had in mind was something like the reply buttons in the email log.

> reply for normal MUAs (but maybe also Delta Chat could show this somewhere)

Makes me think of a tree visualizing lines between the messages in the background... clickable to jump up/down to the referenced message.

**Comment by r10s** [comment]
I think a tree is completely out of scope for now :) 
The reply function would indeed be _mainly_ to stay compatible with normal MUAs, esp. for mailing list where I do not want to be Delta as the client who always replies in a bad way :)

**Comment by csb0730** [comment]
I hint to the message from r10s from 7 Mar.

> Tapping on the Reply button would select the message for a reply, this can be shown as an hint somewhere

A tree isn't necessary IMHO. I think it's sufficient to show the new message just below the related one (maybe in a new colour?).

**Comment by testbird** [comment]
Could work more nicely, with auto-jumping to the oldest unread message when opening the chat, and with some way (button?) at the end of the message to jump to the next unread message if there is one.

(The above could be equally useful for the unapproved emails.)

And again, at least in the mailing list and group chats, the messages could display the reply buttons right in the messages, just as the messages in the unapproved emails, to make adequate in context replies easy.)

**Comment by csb0730** [comment]
Idea: Feed mailing list emails to a special chat like _Contact Requests_(Mailing List Emails). So user has the chance to read them at least :)

**Comment by testbird** [comment]
Sorting into chats is the first ToDo implementation above, but releasing it without also  being able to send  standard email replies isn't a good idea (would lead to a bad reputation).

**Comment by testbird** [comment]
For a natural and intuitive subject handling UI / UX, see: https://support.delta.chat/t/subject-of-emails/264/61

**Comment by csb0730** [comment]
@r10s - Can we simply introduce a user option to at least show list mails? Maybe simply in a way like my request from 17 Aug 2018?

I would be very happy when having this because now, I often miss messages because I'm using DC as the first mail app and only switch to MUA if necessary (rarely).
